pyvista/core/filters/image_data.py

```diff
@@ -430,7 +430,7 @@ def image_threshold(
             array_dtype := self.active_scalars.dtype,
             int,
         ) and array_dtype != np.dtype(np.uint8)
-        cast_dtype = has_int_dtype and platform.system() in ['Darwin', 'Linux']
+        cast_dtype = has_int_dtype and platform.system() in ['Darwin', 'Linux', 'Windows']
```

Comment:
@user27182 I am neither a Windows user nor a a user of this functionality.  Windows fails on this test with numpy 2.0.  Can we get rid of this platform check altogether (I would get rid of the whole check in this case, but did this just to get the CI green).  It seems like it is safest to do the type conversion no matter the platform, and then the behavior will also be consistent.  Thoughts?

Reply:
Sure let's get rid of it this check and always convert to be safe. The checks were conservative and made just to pass our specific CI builds for that PR at the time. But clearly the issue is broader in scope.